### PR TITLE
Few extra racer.c matches

### PR DESCRIPTION
**Pull request by FazanaJ**

### CODE REVIEW

File: src/racer.c [hunk 1]
```diff
@@ -2273,7 +2519,154 @@ void func_80054110(Object *obj, Object_Racer *racer, s32 updateRate, f32 updateR
...
+        traction = 0.02f;
+    }
+    gCurrentSurfaceType = surfaceType;
+    racer->drift_direction = 0;
+    racer->unk10C = 0;
+    if (surfaceType == 4) {
+        multiplier = 0.25f;
+        racer->lateral_velocity = 0.0f;
+    }
+    if (racer->boostTimer == 0 && surfaceType == 3) {
```
Comment (AntonioCastelli) [inline src/racer.c:2590]: Haven't checked in a debugger, but does this imply that `SURFACE_UNK3` is for driving on zippers?

File: src/racer.c [hunk 2]
```diff
@@ -2273,7 +2519,154 @@ void func_80054110(Object *obj, Object_Racer *racer, s32 updateRate, f32 updateR
...
+    }
+    // If basically at a standstill, then allow reversing.
+    if (racer->velocity > -2.0 && racer->velocity < 3.0) {
+        f32 reverseVel = (3.0 - racer->velocity) * 0.15;
+        if (gCurrentStickY < -25 && !(gCurrentRacerInput & A_BUTTON) && gCurrentRacerInput & B_BUTTON) {
+            racer->brake = 0.0f;
+            racer->velocity += reverseVel;
+        }
+    }
+	forwardVel = (vel * racer->brake) * 0.32;
```
Comment (AntonioCastelli) [inline src/racer.c:2641]:
```suggestion
    forwardVel = (vel * racer->brake) * 0.32;
```

File: src/racer.c [hunk 3]
```diff
@@ -1964,12 +2072,150 @@ void racer_spinout_car(Object* obj, Object_Racer* racer, s32 updateRate, f32 upd
...
+    if (racer->unk1FE == 1 && racer->playerIndex == PLAYER_COMPUTER) {
+        gCurrentStickY = 60;
+    }
+    if (racer->unk1F0 != 0) {
+        if (racer->playerIndex != PLAYER_COMPUTER) {
+            steerAngle = racer->steerAngle;
+        } else {
+            steerAngle = gCurrentStickX;
+        }
+        racer->unk1A0 -= (u16) ((steerAngle * 6 * updateRate) >> 1);
```
[the remark this answers is not in the document]
Comment (FazanaJ) [reply src/racer.c:2098]: Nope, neither does changing `>> 1` to `/ 2`

File: src/racer.c (hunk 1 above)
Comment (FazanaJ) [reply src/racer.c]: I assume so, but having tested ingame, I cannot verify for sure, so I'll keep it as UNK3 for now.

**Review by Ryan-Myers — APPROVED** [review]

**Pull request merged**